x-pack/plugins/security/server/routes/api_keys/has.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RouteDefinitionParams } from '..';
+import { wrapIntoCustomErrorResponse } from '../../errors';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+
+/**
+ * Response of Kibana Has API keys endpoint.
+ */
+export interface HasAPIKeysResult {
+  hasApiKeys: boolean;
+}
+
+export function defineHasApiKeysRoutes({
```

Comment:
@azasypkin This API will be consumed in contexts where we want to customize messages and link in the user experience depending on whether they have created API keys. Specifically, the Getting Started flows in serverless will be updated with use of this API.

For the areas I'm updating, it doesn't seem important to rely on `areAPIKeysEnabled`. For serverless, we know that API Keys are enabled. In case of unexpected error, it seems fine to log the error and give the user the default experience they would get when they haven't created API keys.